pkg/kubelet/containerdshim/containerd_container_test.go

```diff
@@ -17,115 +17,192 @@ limitations under the License.
 package containerdshim
 
 import (
-	"fmt"
-	"net"
+	"math/rand"
 	"os"
+	"path/filepath"
 	"testing"
 	"time"
 
-	"google.golang.org/grpc"
-
-	"github.com/docker/containerd/api/services/execution"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	runtimeapi "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+	"k8s.io/kubernetes/pkg/kubelet/dockershim"
 )
 
+func init() {
+	rand.Seed(time.Now().Unix())
+}
+
 // NOTE: To run the test, please make sure `containerd` is running and 'dist' is in $PATH.
 // And you should run the test as root.
-func TestContainer(t *testing.T) {
+func TestContainerOperationError(t *testing.T) {
```

Comment:
Beat me to refactoring this :) I like splitting up the fail bucket and the other changes made thanks!